src/distribution.cpp

```diff
@@ -44,7 +45,7 @@ double Discrete::sample() const
       c += p_[i];
       if (xi < c) return x_[i];
     }
-    // throw exception?
+    throw std::runtime_error{"Error when sampling probability mass function."};
```

Comment:
I'm personally in favor of it. Given that OpenMC is more commonly being used in a library context, it seems better to throw exceptions / return error codes (i.e., something that is recoverable) rather than completely abort. Obviously it's inconsistent at present with a mix of both, but I'd eventually like to get rid of our fatal errors.